onyo/lib/tests/test_git.py

```diff
@@ -0,0 +1,264 @@
+from pathlib import Path
+
+import pytest
+from onyo import OnyoRepo, OnyoInvalidRepoError
+from onyo.lib.git import GitRepo
+
+
+def test_GitRepo_instantiation_existing(repo: OnyoRepo) -> None:
+    """
+    The GitRepo class must instantiate correctly for paths to existing
+    repositories.
+    """
+    new_repo = GitRepo(repo.git.root)
+    assert new_repo.root.samefile(repo.git.root)
+
+
+@pytest.mark.repo_dirs('existing/directory')
+def test_GitRepo_instantiation_find_root(repo: OnyoRepo) -> None:
+    """
+    The GitRepo class must instantiate correctly with paths to sub-directories
+    in a repository when `find_root=True`.
+    """
+    new_repo = GitRepo(repo.git.root / 'existing/directory', find_root=True)
+    assert new_repo.root.samefile(repo.git.root)
+    assert (new_repo.root / '.git').exists()
+
+
+def test_GitRepo_find_root(tmp_path: Path) -> None:
+    """
+    `GitRepo.find_root()` must raise `OnyoInvalidRepoError` when called on a
+    Path which is not a git repository.
+    """
+    # try with an existing but non-repo path
+    with pytest.raises(OnyoInvalidRepoError):
+        GitRepo.find_root(tmp_path)
+    # Path that does not exist at all
+    with pytest.raises(OnyoInvalidRepoError):
+        GitRepo.find_root(Path("This/path/does/not/exist"))
+
+
+@pytest.mark.repo_files('a/test/asset_for_test.0',
+                        'a/test/asset_for_test.1')
+def test_GitRepo_restore_staged(repo: OnyoRepo) -> None:
+    """
+    `GitRepo.restore_staged()` must restore all staged files in the repository.
+    If no files are staged, it should not raise an error.
+    """
+    new_git = GitRepo(repo.git.root)
+    test_file = Path('a/test/asset_for_test.0').resolve()
+    test_file2 = Path('a/test/asset_for_test.1').resolve()
+
+    # test that no error is raised, when called on a clean repository
+    new_git.restore_staged()
+
+    # change two files and add them, so they are "staged for commit"
```

Comment:
If you are only testing whether staged files are restored, what does the second file add?
What kind of bug would it detect? There's nothing different about the second file.
Also: This again tests only part of the behavior. You state "restore all staged files in the repository. If no files are staged, it should not raise an error." What about untracked files or modified files that are not staged? What happens? That's what you could use the second (third, etc.) file for.